### PR TITLE
Make builds thread safe

### LINKED ISSUE #491: cime5.0.9 SMP flag not thread safe in build
Whether a component uses openmp to build is determined by the setting of variable SMP or BUILD_THREADED to the Makefile, SMP is set depending on the value of NTHRDS_* for each component and is currently happening in a non-threadsafe way so if any component may set 
SMP=FALSE then that setting may be interfering with the build of other components.

A workaround is to set the BUILD_THREADED flag in your case to TRUE which will override the SMP setting and just build all components with openmp.

### PR DESCRIPTION
**Pull request by jgfouca**
... by moving environment changes to shell command.

Test suite: create_test --no-run cime_developer
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #491

User interface changes?: No

Code review: jedwards

### CODE REVIEW

File: utils/python/CIME/build.py [hunk 1]
```diff
@@ -523,17 +520,17 @@ def build_libraries(case, exeroot, caseroot, cimeroot, libroot, mpilib, lid, mac
             if (not os.path.isdir(ndir)):
                 os.makedirs(ndir)
 
-        _build_model_thread(config_lnd_dir, "lnd", caseroot, bldroot, libroot, incroot, file_build, logs)
+        _build_model_thread(config_lnd_dir, "lnd", caseroot, bldroot, libroot, incroot, file_build, logs, False)
```
Comment (jgfouca) [inline utils/python/CIME/build.py]: How does lnd know if it's threaded or not?

Comment (jedwards4b) [inline utils/python/CIME/build.py]: Actually - how do any of the shared libraries know?

Comment (jgfouca) [inline utils/python/CIME/build.py]: The global os.environ[SMP] I assume

Comment (jgfouca) [inline utils/python/CIME/build.py]: The thing we were saving as "overall_smp"

Comment (jedwards4b) [inline utils/python/CIME/build.py]: But the variable THREADDIR in Makefile is only set if openmp is on and this appears to be working for shared_libraries.

**Pull request merged**